provider/lib/utils.js

```diff
@@ -310,7 +310,7 @@ module.exports = function(
                     logger.info(method, dataTrigger.id, 'http post request, STATUS:', response ? response.statusCode : response);
                     if (error || response.statusCode >= 400) {
                         logger.error(method, 'there was an error invoking', dataTrigger.id, response ? response.statusCode : error);
-                        if (!error && [408, 429, 500, 502, 503].indexOf(response.statusCode) === -1) {
+                        if (!error && [408, 429, 500, 502, 503, 504].indexOf(response.statusCode) === -1) {
```

Comment:
Does it make sense to separate retry policy from the delete and disable policy? When there is a systemic failure it will affect more than one trigger surely and there is likely a more general solution for retry in that case. Where's as to delete and disable its a much more specific and targeted solution.